test/MessageTraitTest.php

```diff
@@ -215,40 +233,40 @@ public function testWithHeaderReplacesDifferentCapitalization()
     /**
      * @dataProvider invalidGeneralHeaderValues
      */
-    public function testWithAddedHeaderRaisesExceptionForNonStringNonArrayValue($value)
+    public function testWithAddedHeaderRaisesExceptionForNonStringNonArrayValue($value): void
     {
         $this->expectException(InvalidArgumentException::class);
         $this->expectExceptionMessage('must be a string');
 
         $this->message->withAddedHeader('X-Foo', $value);
     }
 
-    public function testWithoutHeaderDoesNothingIfHeaderDoesNotExist()
+    public function testWithoutHeaderDoesNothingIfHeaderDoesNotExist(): void
     {
         $this->assertFalse($this->message->hasHeader('X-Foo'));
         $message = $this->message->withoutHeader('X-Foo');
         $this->assertNotSame($this->message, $message);
         $this->assertFalse($message->hasHeader('X-Foo'));
     }
 
-    public function testHeadersInitialization()
+    public function testHeadersInitialization(): void
     {
         $headers = ['X-Foo' => ['bar']];
         $message = new Request(null, null, 'php://temp', $headers);
         $this->assertSame($headers, $message->getHeaders());
     }
 
-    public function testGetHeaderReturnsAnEmptyArrayWhenHeaderDoesNotExist()
+    public function testGetHeaderReturnsAnEmptyArrayWhenHeaderDoesNotExist(): void
     {
         $this->assertSame([], $this->message->getHeader('X-Foo-Bar'));
     }
 
-    public function testGetHeaderLineReturnsEmptyStringWhenHeaderDoesNotExist()
+    public function testGetHeaderLineReturnsEmptyStringWhenHeaderDoesNotExist(): void
     {
         $this->assertEmpty($this->message->getHeaderLine('X-Foo-Bar'));
     }
 
-    public function headersWithInjectionVectors()
+    public function headersWithInjectionVectors(): array
```

Comment:
After local inspection, the reason why the psalm issues did arise is that all methods were given a new signature, instead of just those related to the fix: currently on it.